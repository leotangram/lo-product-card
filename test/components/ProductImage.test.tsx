import React from 'react';
import renderer from 'react-test-renderer';
import ProductCard, { ProductImage } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductTitle', () => {
  test('should show the component correctly with personalized image', () => {
    const wrapper = renderer.create(
      <ProductImage img="https://cdn.pixabay.com/photo/2015/04/23/22/00/tree-736885__480.jpg" />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should show the component with product image', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
