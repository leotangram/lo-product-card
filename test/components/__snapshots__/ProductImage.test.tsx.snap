// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle should show the component correctly with personalized image 1`] = `
<img
  alt="Product"
  className="productImg "
  src="https://cdn.pixabay.com/photo/2015/04/23/22/00/tree-736885__480.jpg"
/>
`;

exports[`ProductTitle should show the component with product image 1`] = `
<div
  className="productCard "
>
  <img
    alt="Product"
    className="productImg "
    src="./coffee-mug.png"
  />
</div>
`;
